scripts/check-rng-hygiene.sh

```
@@ -0,0 +1,221 @@
+#
+# What this does NOT cover, stated so nobody reads a green run as more than it
+# is: it is a grep. It catches the shape, not the intent -- it cannot tell
+# whether a checked draw is used correctly once generated, nor whether a value
+# reaches the right place. It does not inspect dependencies, so a crate that
+# degrades its own RNG internally is invisible here. Rule 4 checks that the gate
+# is called, not that the health check itself is sound.
+#
+# Run from anywhere. Exits non-zero with the offending lines.
+
+set -uo pipefail
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)